lib/experimental/new_embed.dart

```diff
@@ -245,81 +268,114 @@ class NewEmbed {
   }
 
   void _displayIssues(List<AnalysisIssue> issues) {
-    int errorCount = 0;
-    int otherCount = 0;
-
-    for (AnalysisIssue issue in issues) {
-      if (issue.kind == 'error') {
-        errorCount++;
-      } else if (issue.kind == 'info' || issue.kind == 'warning') {
-        otherCount++;
-      }
+    analysisResultBox.hide();
+    testResultBox.hide();
+
+    if (issues.isEmpty) {
+      return;
     }
 
-    if (errorCount == 0 && otherCount == 0) {
-      analysisResultBox.hide();
-    } else {
-      String message;
-      FlashBoxStyle style;
-
-      if (errorCount > 0 && otherCount > 0) {
-        message = 'Analyzer found $errorCount error${errorCount > 1 ? 's' : ''}'
-            ' and $otherCount other issue${otherCount > 1 ? 's' : ''}.';
-        style = FlashBoxStyle.error;
-      } else if (errorCount > 0) {
-        message =
-            'Analyzer found $errorCount error${errorCount > 1 ? 's' : ''}.';
-        style = FlashBoxStyle.error;
-      } else {
-        message =
-            'Analyzer found $otherCount issue${otherCount > 1 ? 's' : ''}';
-        style = FlashBoxStyle.warn;
+    List<String> messages = issues.map((AnalysisIssue issue) {
+      String message = issue.message;
+      if (message.endsWith('.')) {
+        message = message.substring(0, message.length - 1);
       }
+      return '$message - line ${issue.line}';
+    }).toList();
 
-      analysisResultBox.showStrings([message], style);
-    }
+    analysisResultBox.showStrings(messages, FlashBoxStyle.warn);
   }
 
   /// Perform static analysis of the source code.
-  void _performAnalysis(_) async {
+  void _performAnalysis([_]) {
     _debounceTimer.invoke(() {
       final dartServices = deps[DartservicesApi] as DartservicesApi;
-      final input = SourceRequest()..source = userCodeEditor.document.value;
-      final lines = Lines(input.source);
-      final request = dartServices.analyze(input).timeout(serviceCallTimeout);
+      final userSource = context.dartSource;
+      // Create a synthesis of the user code and other code to analyze.
+      final fullSource = '$userSource\n'
+          '${context.testMethod}\n'
+          '${executionSvc.testResultDecoration}\n'
+          'var resultFunction = _result;\n';
```

Comment:
What is this line doing? I don't see the string "resultFunction" anywhere else in the PR, so I thought I'd ask.

Reply:
Without this, the `_result` function can cause a 'method not used' analysis issue to show. This ensures that we have at least one use of it. I'll make a comment.

Reply:
Got it. Thanks.